python/feathub/registries/mysql_registry.py

```diff
@@ -0,0 +1,289 @@
+        self.tables: Dict[str, Tuple[TableDescriptor, TableDescriptor, datetime]] = {}
+
+        self.conn = mysql.connector.connect(
+            host=self.host,
+            port=self.port,
+            user=self.username,
+            password=self.password,
+            database=self.database,
+        )
+        self.cursor = self.conn.cursor()
+
+        self.cursor.execute(
+            f"""
+                CREATE TABLE IF NOT EXISTS `{self.table}`(
+                   `name` VARCHAR(64) NOT NULL,
+                   `timestamp` TIMESTAMP NOT NULL,
+                   `digest` VARCHAR(64) NOT NULL,
+                   `original_descriptor` TEXT NOT NULL,
+                   `resolved_descriptor` TEXT NOT NULL,
+                   PRIMARY KEY ( `name`, `timestamp` )
+                );
+            """
+        )
+
+    def build_features(
+        self,
+        feature_descriptors: List[TableDescriptor],
+        force_update: bool = False,
+        props: Optional[Dict] = None,
+    ) -> List[TableDescriptor]:
+        result = []
+        for table in feature_descriptors:
+            if table.name == "":
+                raise FeathubException(
+                    "Cannot build a TableDescriptor with empty name."
+                )
+
+            # TODO: add document about the limitations on the length of feature
+            #  table names.
+            if len(table.name) > 64:
+                raise FeathubException(
+                    "Cannot build or register a descriptor with a name longer"
+                    "than 64 characters."
+                )
+
+            self.tables[table.name] = (
+                table,
+                table.build(self, force_update=force_update, props=props),
+                datetime.utcnow(),
+            )
+            result.append(self.tables[table.name][1])
+
+        return result
+
+    def register_features(
+        self, feature_descriptors: List[TableDescriptor], force_update: bool = False
+    ) -> bool:
+        self.build_features(feature_descriptors, force_update=force_update)
+
+        for descriptor in feature_descriptors:
+            existing_descriptor = self._get_descriptor_from_mysql(descriptor.name)
+
+            original_descriptor = json.dumps(
+                self.tables[descriptor.name][0].to_json(), sort_keys=True
+            )
+            resolved_descriptor = json.dumps(
+                self.tables[descriptor.name][1].to_json(), sort_keys=True
+            )
+            digest = _get_digest(original_descriptor, resolved_descriptor)
+
+            if existing_descriptor is not None and existing_descriptor[2] == digest:
+                continue
+
+            original_descriptor = original_descriptor.replace('"', '\\"')
+            resolved_descriptor = resolved_descriptor.replace('"', '\\"')
+            timestamp = self.tables[descriptor.name][2].strftime("%Y-%m-%d %H:%M:%S")
+            self.cursor.execute(
```

Comment:
Can we compare the timestamp and only insert an entry if the timestamp in the mysql is older?